File: nim/str_join_slow.nim
import times
import strutils

const chunk = "regexp123abcbcd"
const repeatCount = 10000
const size = 1000

proc join(): void =
    var chunks: seq[string] = @[]
    for i in 1..size:
        chunks.add(chunk)
    discard chunks.join("")

let startTime = epochTime()
for i in 1..repeatCount:
    join()
let endTime = epochTime()

echo int((endTime - startTime) * 1000)

File: perl/str_join_slow.pl
use Time::HiRes qw/gettimeofday tv_interval/;

my $chunk = "regexp123abcbcd";
my $repeat_count = 10000;
my $size = 1000;

sub strjoin {
  my @chunks = ();
  $chunks[$size - 1] = $chunk;
  for (my $i = 0; $i < $size - 1; ++$i) {
    $chunks[$i] = $chunk;
  }
  join "", @chunks;
}

my $start = [gettimeofday];
for (my $i = 0; $i < $repeat_count; ++$i) {
  strjoin;
}
my $end = [gettimeofday];
print int(tv_interval($start, $end) * 1000), "\n";
